pipeline-runner/src/doubletScores.r

```diff
@@ -7,14 +7,13 @@
 
 source('utils.r')
 
-# The most uses values in doublet scores reporting in the scrublet paper [1] are around 0.25. There are not too much literature about how to compute
-# a threshold. For now, we will offer two methods:
-# --> Absolute threshold: In order to be not too extrictive the threshold is set to 0.25
-generate_default_values_doubletScores <- function(seurat_obj, config) {
-
-    # HARDCODE
-    threshold <- 0.25
-
+generate_default_values_doubletScores <- function(seurat_obj, sample) {
```

Comment:
I would suggest not changing the parameter config to sample, since this function should be the same of [this](https://github.com/biomage-ltd/pipeline/blob/89577460b5ef98babe095a509513bc2c15191d90/pipeline-runner/src/data-ingest/4_Prepare_experiment.r#L31-L39) (Moreover, in the data-ingest it seems that it was updated for scDblFinder, but in the pipeline it wasn't)

Reply:
The function you linked is being called [here](https://github.com/biomage-ltd/data-ingest/blob/0fbeb48c9eda8b9b1cf08e009ab3551a12af17ca/src/4_Prepare_experiment.r#L336) which first [subsets](https://github.com/biomage-ltd/pipeline/blob/89577460b5ef98babe095a509513bc2c15191d90/pipeline-runner/src/data-ingest/4_Prepare_experiment.r#L94) the suerat object so that only cells for a given sample are present.

In contrast here the seurat object is not a subset so we need the sample name in order to generate the appropriate per sample threshold when the user has selected _Automatic_. I don't beleive the `config` parameter was even used previously in this function.

Reply:
Originally, we had a config function per step, where the structure was going to have `seurat_obj` and `config` as parameter of the functions. However, in the end it was not realized for all steps. So, I leave it up to you to filter inside the function or outside :)

I think the rest of the changes are all good, when we can test it in staging, I am happy to approve.

Reply:
staging link added. I uploaded with Vicky's dataset without obvious issues.